nix/nix.go

```diff
@@ -44,14 +44,19 @@ func Shell(path string) error {
 	//
 	// ORIGINAL_PATH is set by sh.StartCommand.
 	// PURE_NIX_PATH is set by the shell hook in shell.nix.tmpl.
-	_ = sh.SetInit(`
-# Update the $PATH so the user can keep using programs that live outside of Nix,
-# but prefer anything installed by Nix.
-export PATH="$PURE_NIX_PATH:$ORIGINAL_PATH"
+	sh.PreInitHook = `
+# Update the $PATH so that the user's init script has access to all of their
```

Comment:
Should both the PreInit and PostInit hooks set the path to: `$PURE_NIX_PATH:$ORIGINAL_PATH`?

Reply:
I just switched both hooks to use the Nix packages.

I wasn't sure what the correct behavior should be:

- Prefer the non-devbox PATH to guarantee compatibility with the user's rcfile.
- Prefer the devbox PATH so that hooks use the Nix packages.